compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/PersistRecordValidator.java

```diff
@@ -180,12 +177,24 @@ private Optional<AnnotationNode> getEntityAnnotation(NodeList<AnnotationNode> an
         return Optional.empty();
     }
 
-    private boolean isDuplicateEntity(Entity entity, TypeDefinitionNode typeDefinitionNode) {
-        if (this.entities.containsKey(entity.getEntityName())) {
-            String initialModule = this.entities.get(entity.getEntityName()).getModule();
-            entity.addDiagnostic(typeDefinitionNode.typeName().location(), DiagnosticsCodes.PERSIST_119.getCode(),
-                    MessageFormat.format(DiagnosticsCodes.PERSIST_119.getMessage(), initialModule),
-                    DiagnosticsCodes.PERSIST_119.getSeverity());
+    private void validateEntityInMultipleSubModule(Entity entity) {
+        if (this.isEntityInMultipleSubModules) {
+            entity.addDiagnostic(entity.getLocation(), DiagnosticsCodes.PERSIST_119.getCode(),
+                    DiagnosticsCodes.PERSIST_119.getMessage(), DiagnosticsCodes.PERSIST_119.getSeverity());
+        } else if (entityInMultipleSubModules(entity)) {
+            for (Map.Entry<String, Entity> entry : this.entities.entrySet()) {
+                entity.addDiagnostic(entry.getValue().getLocation(), DiagnosticsCodes.PERSIST_119.getCode(),
+                        DiagnosticsCodes.PERSIST_119.getMessage(), DiagnosticsCodes.PERSIST_119.getSeverity());
+            }
+            entity.addDiagnostic(entity.getLocation(), DiagnosticsCodes.PERSIST_119.getCode(),
+                    DiagnosticsCodes.PERSIST_119.getMessage(), DiagnosticsCodes.PERSIST_119.getSeverity());
+        }
+    }
+
+    private boolean entityInMultipleSubModules(Entity entity) {
+        String entityLocation = this.entities.get(this.entities.keySet().iterator().next()).getModule();
+        if (!entityLocation.equals(entity.getModule())) {
+            this.isEntityInMultipleSubModules = true;
```

Comment:
Isnt it simpler to keep the module name of the first entity defined as a global variable and compare the submodule each time we validate its an entity. Rather than this?